src/exports.cc

```diff
@@ -872,6 +885,351 @@ Word wasi_unstable_clock_time_get(Word clock_id, uint64_t /*precision*/,
   return 0; // __WASI_ESUCCESS
 }
 
+// __wasi_errno_t __wasi_clock_res_get(__wasi_clockid_t id, __wasi_timestamp_t *retptr0);
+Word wasi_unstable_clock_res_get(Word clock_id, Word result_resolution_uint64_ptr) {
+  uint64_t result = 0;
+  auto *context = contextOrEffectiveContext();
+  
+  switch (clock_id) {
+  case 0 /* realtime */:
+  case 1 /* monotonic */:
+    // Return 1 nanosecond as the resolution for both realtime and monotonic clocks
+    // This is a reasonable default for most systems
+    result = 1;
+    break;
+  default:
+    // process_cputime_id and thread_cputime_id are not supported yet.
+    return 58; // __WASI_ENOTSUP
+  }
+  
+  if (!context->wasm()->setDatatype(result_resolution_uint64_ptr, result)) {
+    return 21; // __WASI_EFAULT
+  }
+  return 0; // __WASI_ESUCCESS
+}
+
+// __wasi_errno_t __wasi_fd_advise(__wasi_fd_t fd, __wasi_filesize_t offset, __wasi_filesize_t len, __wasi_advice_t advice);
+Word wasi_unstable_fd_advise(Word fd, uint64_t offset, uint64_t len, Word advice) {
+  // fd_advise is used to provide advice about the expected behavior of the application with respect to a file.
+  // Since we don't have a real file system in proxy-wasm, we can just return success without doing anything.
+  // This is similar to how other file-related functions are implemented in this codebase.
+  
+  // We could check if fd is valid (e.g., stdout/stderr), but since this is just a hint and not required
+  // for correctness, we'll just return success for any fd.
+  
+  return 0; // __WASI_ESUCCESS
+}
+
+// __wasi_errno_t __wasi_fd_allocate(__wasi_fd_t fd, __wasi_filesize_t offset, __wasi_filesize_t len);
+Word wasi_unstable_fd_allocate(Word fd, uint64_t offset, uint64_t len) {
+  // fd_allocate is used to ensure that space is allocated for a file.
+  // Since we don't have a real file system in proxy-wasm, we can just return success without doing anything.
+  // This is similar to how other file-related functions are implemented in this codebase.
+  
+  // We only support stdout and stderr in proxy-wasm, which don't need allocation
+  if (fd != 1 /* stdout */ && fd != 2 /* stderr */) {
+    return 8; // __WASI_ERRNO_BADF - Bad file descriptor
+  }
+  
+  return 0; // __WASI_ESUCCESS
+}
+
+// __wasi_errno_t __wasi_fd_datasync(__wasi_fd_t fd);
+Word wasi_unstable_fd_datasync(Word fd) {
+  // fd_datasync is used to synchronize the data of a file to disk.
+  // Since we don't have a real file system in proxy-wasm, we can just return success for stdout/stderr
+  // and an error for other file descriptors.
+  
+  // We only support stdout and stderr in proxy-wasm
+  if (fd != 1 /* stdout */ && fd != 2 /* stderr */) {
+    return 8; // __WASI_ERRNO_BADF - Bad file descriptor
+  }
+  
+  // For stdout and stderr, there's no need to sync as they're handled by the host system
+  return 0; // __WASI_ESUCCESS
```

Comment:
This code is repeated a lot, but are those function actually called for `stdout` and `stderr` or is it simply copy & paste? Some of those would be rather surprising.